src/Microsoft.OData.Core/JsonLight/ODataJsonLightResourceDeserializer.cs

```diff
@@ -1333,13 +3463,14 @@ private ODataJsonLightReaderNestedInfo TryReadAsStream(IODataJsonLightReaderReso
                     isCollection &&
                     propertyType.Definition.AsElementType().IsUntyped()) ||
                 (propertyType == null
-                    && (isCollection || this.JsonReader.CanStream())
+                    && (isCollection || await this.JsonReader.CanStreamAsync().ConfigureAwait(false))
                     && readAsStream != null
                     && readAsStream(null, isCollection, propertyName, property)))

```

Comment:
Should this delegate also be synchronous in the async function (I'm not sure what it does, but it starts with `read` so just wanted to double-check that it doesn't any I/O operation that should be async).

Reply:
@habbes No. The delegate doesn't ready from the stream. It's a `Func` property declared in `ODataMessageReaderSettings` class that someone can use to direct the reader to read particular properties as a stream.